vortex-zigzag/src/compress.rs

```
@@ -1,26 +1,23 @@
-use vortex::array::downcast::DowncastArrayBuiltin;
 use vortex::array::primitive::PrimitiveArray;
-use vortex::array::{Array, ArrayKind, ArrayRef};
 use vortex::compress::{CompressConfig, CompressCtx, EncodingCompression};
 use vortex::ptype::{NativePType, PType};
-use vortex::stats::Stat;
-use vortex::validity::{OwnedValidity, ValidityView};
-use vortex::view::ToOwnedView;
+use vortex::stats::{ArrayStatistics, Stat};
+use vortex::validity::Validity;
+use vortex::{Array, ArrayDef, AsArray, IntoArray, OwnedArray, ToStatic};
 use vortex_alloc::{AlignedVec, ALIGNED_ALLOCATOR};
 use vortex_error::VortexResult;
-use zigzag::ZigZag;
+use zigzag::ZigZag as ExternalZigZag;
 
-use crate::downcast::DowncastZigzag;
-use crate::zigzag::{ZigZagArray, ZigZagEncoding};
+use crate::{OwnedZigZagArray, ZigZag, ZigZagEncoding};
 
 impl EncodingCompression for ZigZagEncoding {
     fn can_compress(
         &self,
-        array: &dyn Array,
+        array: &Array,
         _config: &CompressConfig,
     ) -> Option<&dyn EncodingCompression> {
         // Only support primitive arrays
-        let parray = array.maybe_primitive()?;
+        let parray = array.as_primitive();
```

This now panics instead of returning None